src/org/traccar/database/NotificationManager.java

```diff
@@ -42,12 +44,16 @@ public NotificationManager(DataManager dataManager) {
         geocodeOnRequest = Context.getConfig().getBoolean("geocoder.onRequest");
     }
 
-    private Set<Long> getEffectiveNotifications(long userId, long deviceId) {
+    private Set<Long> getEffectiveNotifications(long userId, long deviceId, Date date) {
```

Comment:
I would rename it to `time` to make it clear that it's a full time, not just date.